account_payment_line/models/account_payment.py

```diff
@@ -82,12 +79,11 @@ def _hook_create_new_line(self, invoice, aml, amount_to_apply):
         )
 
     def action_propose_payment_distribution(self):
-        move_model = self.env["account.move"]
+        domain = self._get_moves_domain()
+        pending_invoices = self.env["account.move"].search(
+            domain, order="invoice_date_due ASC"
+        )
         for rec in self:
-            if self.is_internal_transfer:
-                continue
-            domain = self._get_moves_domain()
```

Comment:
Keep the domain here.
And, yes, it was incorrect now, change it to "domain = `rec`._get_moves_domain()"